perses/rjmc/geometry.py

```diff
@@ -644,15 +648,15 @@ def _logp_propose(self, top_proposal, old_positions, beta, new_positions=None, d
 
         state = final_context.getState(getEnergy=True)
         final_context_reduced_potential = beta*state.getPotentialEnergy()
-        final_context_components = [(force, energy*beta) for force, energy in compute_potential_components(final_context)]
-        atoms_with_positions_reduced_potential_components = [(force, energy*beta) for force, energy in compute_potential_components(atoms_with_positions_context)]
+        final_context_components = [(force, energy*beta) for force, energy in compute_potential_components(final_context, platform=openmm.Platform.getPlatformByName("Reference"))]
+        atoms_with_positions_reduced_potential_components = [(force, energy*beta) for force, energy in compute_potential_components(atoms_with_positions_context, platform=openmm.Platform.getPlatformByName("Reference"))]
         _logger.debug(f"reduced potential components before atom placement:")
         for item in atoms_with_positions_reduced_potential_components:
             _logger.debug(f"\t\t{item[0]}: {item[1]}")
         _logger.info(f"total reduced potential before atom placement: {atoms_with_positions_reduced_potential}")
 
         _logger.debug(f"potential components added from growth system:")
-        added_energy_components = [(force, energy*beta) for force, energy in compute_potential_components(context)]
+        added_energy_components = [(force, energy*beta) for force, energy in compute_potential_components(context, platform=openmm.Platform.getPlatformByName("Reference"))]
```

Comment:
Do we want to switch to the fastest double-precision platform here?

Reply:
So switching to 'CUDA' double-precision for this test, the test still fails which seems to be due to a difference in the `NonbondedForce` when calculated on CUDA or Reference. @jchodera do you know why this might be?

For a _real_ system:
CUDA, double: NonbondedForce: -156986.35768798555
Reference: NonbondedForce: -156986.41156104617


Our energy threshold is small 1E-4 so this is enough to tip it into being "unvalid"